spec/java/util/stream/DoubleStream.main.lsl

```diff
@@ -0,0 +1,1266 @@
+            result = new LongStreamAutomaton(state = Initialized,
+                storage = emptyArray,
+                length = this.length,
+                closeHandlers = this.closeHandlers,
+            );
+        }
+        else
+        {
+            val newStorage: array<long> = action ARRAY_NEW("long", this.length);
+
+            var i: int = 0;
+            action LOOP_FOR(
+                i, 0, this.length, +1,
+                _convertDoubleStorageToLong_loop(i, newStorage)
+            );
+
+            result = new LongStreamAutomaton(state = Initialized,
+                storage = newStorage,
+                length = this.length,
+                closeHandlers = this.closeHandlers,
+            );
+        }
+    }
+
+
+    @Phantom proc _convertDoubleStorageToLong_loop(i: int, newStorage: array<long>): void
+    {
+        newStorage[i] = this.storage[i] as long;
+    }
+
+
+    fun *.asIntStream (@target self: DoubleStream): IntStream
+    {
+        if (this.length == 0)
+        {
+            val emptyArray: array<int> = action ARRAY_NEW("int", 0);
+            result = new IntStreamAutomaton(state = Initialized,
+                storage = emptyArray,
+                length = this.length,
+                closeHandlers = this.closeHandlers,
+            );
+        }
+        else
+        {
+            val newStorage: array<int> = action ARRAY_NEW("int", this.length);
+
+            var i: int = 0;
+            action LOOP_FOR(
+                i, 0, this.length, +1,
+                _convertDoubleStorageToInt_loop(i, newStorage)
+            );
+
+            result = new IntStreamAutomaton(state = Initialized,
+                storage = newStorage,
+                length = this.length,
+                closeHandlers = this.closeHandlers,
+            );
+        }
+    }
+
+
+    @Phantom proc _convertDoubleStorageToInt_loop(i: int, newStorage: array<int>): void
+    {
+        newStorage[i] = this.storage[i] as int;
+    }
+
+
+    fun *.sum (@target self: DoubleStream): double
+    {
+        result = 0;
+
+        if (this.length != 0)
+        {
+            result = _sum();
+        }
+    }
+
+
+    fun *.average (@target self: DoubleStream): OptionalDouble
+    {
+        if (this.linkedOrConsumed)
+            _throwISE();
+
+        if (this.length == 0)
+        {
+            result = action DEBUG_DO("OptionalDouble.empty()");
+        }
+        else
+        {
+            var curSum: double = _sum();
+            var divisionResult: double = curSum / this.length;
+            result = action DEBUG_DO("OptionalDouble.of(divisionResult)");
+        }
+
+        this.linkedOrConsumed = true;
+    }
+
+
+    fun *.summaryStatistics (@target self: DoubleStream): DoubleSummaryStatistics
+    {
+        if (this.linkedOrConsumed)
+            _throwISE();
+
+        // #problem I'm waiting DoubleSummaryStatistics type in separated files
```

Comment:
The type in question have been added as part of efforts on separating type declarations into separate files, as I remember.

Reply:
Fixed;